File: javascript/boj/stack/1874.js
const path = process.platform === 'linux' ? '/dev/stdin' : 'input.txt';
const [n, ...seq] = require('fs').readFileSync(path).toString().trim().split('\n').map(Number);

const answer = [];
const stk = [];
let top = 1;

for (let i = 0; i < n; i++) {
    while (top <= seq[i]) {
        stk.push(top);
        answer.push('+');
        top++;
    }

    if (stk[stk.length-1] == seq[i]) {
        stk.pop();
        answer.push('-');
    } else {
        console.log('NO');
        return;
    }
}

for (const ans of answer) {
    console.log(ans);
}

File: python/boj/1874.py
answer = []
stk = []
top = 1

for _ in range(int(input())):
  seq = int(input())
  
  while top <= seq:
    stk.append(top)
    answer.append('+')
    top += 1

  if stk[-1] == seq:
    stk.pop()
    answer.append('-')
  else:
    print('NO')
    exit(0)				

for ans in answer:
  print(ans)